doc/README.yaml.md

```diff
@@ -0,0 +1,230 @@
+
+# README for Open-CE build YAML files
+
+
+This README documents the YAML files used by the build scripts
+[build_env.py](https://github.com/open-ce/open-ce/blob/master/doc/README.build_env.md)
+and [build_feedstock.py](https://github.com/open-ce/open-ce/blob/master/doc/README.build_feedstock.md).
+Please refer to the respective README files for each of these two scripts.
+
+Most likely you will want to simply use the YAML files provided within the
+open-ce repository as they are. These files have already been customized and
+verified to work. However, if you want to debug or do your own customizations,
+you can edit them or create your own YAML files for use by these scripts. If you
+do this, you will likely want to start by copying one of the existing files and
+making local edits as desired.
+
+NOTE: This file only documents information about the YAML files as used within
+the open-ce project. This file is nowhere near being a full reference of the
+YAML configuration markup language. Only fields of interest to open-ce are noted
+here. For more detail or education about YAML files in general, refer to the
+public YAML specifications at https://yaml.org/ or in any number of independent
+guides available elsewhere online.
+
+
+# build_env environment YAML file
+
+The `build_env.py` command uses what we call an environment YAML file.
+In this context, an "environment" is a fully-built core package along with all
+of its dependencies, each of which is found in a separate "feedstock" repository
+within the broader open-ce project.
+Default YAML files are provided for each of several environments in the open-ce
+repository, within https://github.com/open-ce/open-ce/tree/master/envs
+These files are named for their package environments, e.g.
+[tensorflow-env.yaml](https://github.com/open-ce/open-ce/blob/master/envs/tensorflow-env.yaml)
+or [pytorch-env.yaml](https://github.com/open-ce/open-ce/blob/master/envs/pytorch-env.yaml),
+among others. You can review these files as examples.
+
+Each environment file will contain a "packages" stanza which will list a number
+of feedstock repositories that are needed to build as requisites. Each feedstock
+dependency is listed on a separate line in the stanza. For example, in
+[xgboost-env.yaml](https://github.com/open-ce/open-ce/blob/master/envs/xgboost-env.yaml):
+```
+packages:
+  - feedstock : nccl              #['cuda' in build_type]
```

Comment:
This should now be `==` not `in`.
https://github.com/open-ce/open-ce/blob/master/envs/xgboost-env.yaml
```suggestion
  - feedstock : nccl              #[build_type == 'cuda']
```